src/utils/network-monitor.ts

```diff
@@ -1128,6 +1162,35 @@ export class NetworkMonitor {
 
       this.blockJobHandler(job.network)
     }
+
+    // update last processed block
+    if (this.apiService !== undefined) {
```

Comment:
Maybe make this whole thing into one function. So it can either be 1 line or at worse a try catch.

```
// update last process block
try {
  //call update block
catch (e) {
  //some log
}
```


I would also maybe move the logic after the log that says its done processing.